packages/dd-trace/src/plugins/tracing.js

```diff
@@ -55,7 +55,23 @@ class TracingPlugin extends Plugin {
   start () {} // implemented by individual plugins
 
   finish () {
-    this.activeSpan?.finish()
+    const span = this.activeSpan
+    if (span) {
+      this.tagBaseService(span)
+      span.finish()
```

Comment:
Are the plugins the only ones that modifies the service name? or a user could set the service name when using the `tracer.trace()` sdk method (and maybe others I don't know well the api :P)?

Moving `tagBaseService` to `opentracing/span.js` and calling it from `DatadogSpan.finish()` wouldn't cover all possible cases? or may the performance be the key...

Reply:
Good point! I'd like to avoid computing tags this low in the internal architecture (which is why this is in plugins), but indeed it doesn't cover users instrumenting by themselves.

I have 2 other interesting spots to add it:
* `encode/tags-processor` - in its current state this only does normalization
* `opentracing/tracer:startSpan`

The tracer route seems more fitting to me, WDYT?

Reply:
`opentracing/tracer:startSpan` sounds good to me

Reply:
umm, after reviewing it, I'm not sure.

Because it seems that `opentracing/tracer:startSpan` is always setting a fixed `service.name` in the span. And the users service names are setted after calling `startSpan` at `tracer.js:addTags`? but I could be wrong :)

Reply:
🤦 indeed. This makes the tag processor the best place for this, despite the fact that what I'm doing is not normalization.